USAGE.md

```diff
@@ -3,34 +3,80 @@
 ## 1. Vault Enterprise License
 Retrieve a [Vault Enterprise trial license](https://www.hashicorp.com/products/vault/trial).
 
-Export the license in your shell environment:
-```yaml
-export VAULT_LICENSE=02MV4UU..
-```
-
-## 2. Start Containers
+## 2. Setup environment
+Export environment variables in a shell for
+
+* the Vault license
+  ```vash
+  export VAULT_LICENSE=02MV4UU..
+  ```
+* the runtime you're going to use
+  ```bash
+  export CONTAINER_RUNTIME=${CONTAINER_RUNTIME:-docker}
+  ```
+  *The default is `docker`. If podman is going to be used, change this to `podman`.*
+* the internal domain containers can be addressed with
+  ```bash
+  if [[ -z ${CONTAINER_RUNTIME##docker} ]];then export CONTAINER_DOMAIN=docker.internal; else export CONTAINER_DOMAIN=dns.podman; fi
+  ```
+  *uses `docker.internal` if docker is going to be used and `dns.podman` if podman. This does not need modification unless another domain name is required.*
+* the realm domain for the freeipa server
+  ```bash
+  export IPA_REALM=${CONTAINER_DOMAIN^^}
+  ```
+
+Or create a `.env` file for `(docker|podman)-compose`
+```bash
+# Example for podman
+CONTAINER_RUNTIME=podman
+VAULT_LICENSE=02MV4UU..
+echo CONTAINER_RUNTIME=${CONTAINER_RUNTIME:-docker} > .env
+if [[ -z ${CONTAINER_RUNTIME##docker} ]];then echo CONTAINER_DOMAIN=docker.internal; else echo CONTAINER_DOMAIN=dns.podman; fi >> .env
+echo VAULT_LICENSE=${VAULT_LICENSE} >> .env
+echo IPA_REALM=${CONTAINER_DOMAIN^^} >> .env
+```
+
+## 3. Start Containers
 
 Start FreeIPA, Keycloak, Vault and K3s:
 ```bash
-docker-compose up -d
+sudo ${CONTAINER_RUNTIME}-compose up -d
```

Comment:
Pod man is user separated; you run all commands with sudo or none.

Reply:
exactly. It's required for the networking to get a proper bridge. In user context, CNI is used. This works but it is problematic with the freeipa pseudo container. I didn't find a good way around that. Since it is a playground and shouldn't be used in production, I think that's fine.

Reply:
sure, consider this solved then. I don't mind having the `sudo` part in front of all the commands in the instructions, I'll just skip them in my playground environment.